continuous-integration/readme.md

```diff
@@ -1,3 +1,227 @@
+
+### IAC DevOPS: Operations by Pull Request
+
+- The Infrastructure deployment process built around a repo that holds the current expected state of the system / Azure environment.
+- Operational changes are made to the running system by making commits on this repo.
+- Git also provides a simple model for auditing deployments and rolling back to a previous state.
+
+### Infrastructure Advocated Patterns
+
+- You define infrastructure as code in Terraform / ARM / Ansible templates
+- Templates are repeatable cloud resource stacks with a focus on configuration sets aligned with app scaling and throughput needs.
+
+### IAC Principles
+
+- [ ] **Automate the Azure Environment**
+  - All cloud resources are provisioned through a set of infrastructure as code templates. This also includes secrets, service configuration settings, role assignments and monitoring conditions.
+  - Azure Portal should provide a read-only view on environment resources. Any  change applied to the environment should be made through the IAC CI tool-chain only.
+  - Provisioning cloud environments should be a repeatable process that's driven off the infrastructure code artifacts checked into our git repository.
+- [ ] **IAC CI Workflow**
+  - When the IAC template files change through a git-based workflow, A CI build pipeline builds, validates and reconciles the target infrastructure environment's current state with the expected state. The infrastructure execution plan candidate for these fixed environments are reviewed by an cloud administer as a gate check prior to the deploy stage of the pipeline applying the execution plan.
+- [ ] **Developer Read-Only Access to Cloud Resources**
+  - Developer accounts in active directory should have read-only access to fixed environments in Azure. Developers can interface with these environments through a system generated service principal which has read/write access to provisioned cloud resources.
+- [ ] **Secret Automation**
+  - Secrets and configuration changes should only be carried out through the IAC devops workflow. In summary, secret management should be automated and repeatable to promote consistency across environments and developer-specific test environments.
+- [ ] **Infrastructure Integration Test Automation**
+  - End-to-end integration tests are run as part of your IAC CI process to inspect and validate that an azure environment is ready for use.
+- [ ] **Infrastructure Documentation**
+  - The deployment and cloud resource template topology should be documented and well understood within the README of the IAC git repo.
+  - Local environment and CI workflow setup steps should be documented.
+
+## Integration Validation
+
+An effective way to identify bugs in your build at a rapid pace is to invest early into a reliable suite of automated tests that validate the baseline functionality of the system:
+
+- [ ] **End to end integration tests**
```

Comment:
In some cases, e2e and integration tests take quite some time to run and therefore impact velocity. We could suggest alternatives that run lenghty validations through timers or only upon merge to master.

Reply:
Added a bullet to provide more context with `Keeping the build fast`